pkg/cmd/roachtest/cluster.go

```diff
@@ -1393,6 +1394,61 @@ func (c *clusterImpl) FetchDebugZip(
 	})
 }
 
+// FetchVMSpecs downloads the VM specs from the cluster using `roachprod get`.
+// The logs will be placed in the test's artifacts dir.
+func (c *clusterImpl) FetchVMSpecs(ctx context.Context, l *logger.Logger) error {
+	if c.IsLocal() {
+		return nil
+	}
+
+	l.Printf("fetching VM specs\n")
+	c.status("fetching VM specs")
+
+	vmSpecsFolder := filepath.Join(c.t.ArtifactsDir(), "vm_specs")
+	if err := os.MkdirAll(vmSpecsFolder, 0755); err != nil {
+		return err
+	}
+
+	// Don't hang forever if we can't fetch the specs.
+	return timeutil.RunWithTimeout(ctx, "fetch logs", 5*time.Minute, func(ctx context.Context) error {
+		cachedCluster, err := getCachedCluster(c.name)
+		if err != nil {
+			return err
+		}
+		providerToVMs := bucketVMsByProvider(cachedCluster)
+
+		for provider, vms := range providerToVMs {
+			p := vm.Providers[provider]
+			vmSpecs, err := p.GetVMSpecs(l, vms)
+			if err != nil {
+				l.Errorf("failed to get VM spec for provider %s: %s", provider, err)
+				continue
+			}
+			for _, vmSpec := range vmSpecs {
+				name, ok := vmSpec["name"].(string)
+				if !ok {
+					l.Printf("failed to create spec files for some VMs")
```

Comment:
Nit: `l.Errorf`

Reply:
Also, if we were to ever hit this branch, it would be helpful to include `vmSpec` (or at least `vmSpec["name"]`) to make it debuggable.